nautobot/dcim/views.py

```diff
@@ -796,545 +1002,12 @@ class DeviceTypeUIViewSet(NautobotUIViewSet):
-                                icon="mdi-trash-can-outline",
-                                size="xs",
-                                form_id="devicebaytemplate_form",
-                                weight=400,
-                            ),
-                        ],
-                    ),
-                ),
-            ),
-            object_detail.DistinctViewTab(
-                weight=900,
-                tab_id="modulebays",
-                label="Module Bays",
-                url_name="dcim:devicetype_modulebays",
-                related_object_attribute="module_bay_templates",
-                panels=(
-                    object_detail.ObjectsTablePanel(
-                        section=SectionChoices.FULL_WIDTH,
-                        weight=100,
-                        table_title="Module Bays",
-                        table_class=tables.ModuleBayTemplateTable,
-                        table_filter="device_type",
-                        tab_id="modulebays",
-                        enable_bulk_actions=True,
-                        form_id="modulebaytemplate_form",
-                        footer_buttons=[
-                            object_detail.FormButton(
-                                link_name="dcim:modulebaytemplate_bulk_rename",
-                                link_includes_pk=False,
-                                label="Rename",
-                                color=ButtonActionColorChoices.RENAME,
-                                icon="mdi-pencil",
-                                size="xs",
-                                form_id="modulebaytemplate_form",
-                                weight=200,
-                            ),
-                            object_detail.FormButton(
-                                link_name="dcim:modulebaytemplate_bulk_edit",
-                                link_includes_pk=False,
-                                label="Edit",
-                                color=ButtonActionColorChoices.EDIT,
-                                icon="mdi-pencil",
-                                size="xs",
-                                form_id="modulebaytemplate_form",
-                                weight=300,
-                            ),
-                            object_detail.FormButton(
-                                link_name="dcim:modulebaytemplate_bulk_delete",
-                                link_includes_pk=False,
-                                label="Delete",
-                                color=ButtonActionColorChoices.DELETE,
-                                icon="mdi-trash-can-outline",
-                                size="xs",
-                                form_id="modulebaytemplate_form",
-                                weight=400,
-                            ),
-                        ],
-                    ),
-                ),
+                children=tuple(make_add_component_button(*cfg) for cfg in ADD_COMPONENTS_CONFIG),
```

Comment:
This is a few more levels of indirection than I was thinking, but it's *probably* OK?